test/ctia/test_helpers/search.clj

```diff
@@ -152,95 +159,140 @@
-                            :short-id
-                            (format "ctia/%s/%s" entity))]
-        (delete delete-uri
-                :headers {"Authorization" "45c1f5e3f05d0"})))))
+    (doseq [full-id (concat matched-ids unmatched-ids)]
+      (delete-doc entity full-id))))
+
+(defn ensure-one-document
+  [f example entity & args]
+  (let [{{full-id :id} :parsed-body} (create-doc entity (dissoc example :id))]
+       (apply (partial f entity) args)
+       (delete-doc entity full-id)))
 
 (defn test-non-describable-search
   [entity query query-field]
-  (let [search-uri (format "ctia/%s/search" (name entity))]
-    (testing (format "GET %s" search-uri)
-      (let [response (search search-uri query)]
-        (is (= 200 (:status response)))
-        (is (= query (first (map query-field (:parsed-body response))))
-            "query term works"))
+  (testing "search term filter"
+    (let [response (search-text entity query)]
+      (is (= 200 (:status response)))
+      (is (= query (first (map query-field (:parsed-body response))))
+          "query term works"))
 
-      (with-redefs [log* (fn [& _] nil)]
-        ;; avoid unnecessary verbosity
-        (let [response (search search-uri "2607:f0d0:1002:0051:0000:0000:0000:0004")]
-          (is (= 400 (:status response)))))
-
-      (let [response (get search-uri
-                          :headers {"Authorization" "45c1f5e3f05d0"}
-                          :query-params {"query" query
-                                         "tlp" "red"})]
-        (is (= 200 (:status response)))
-        (is (empty? (:parsed-body response))
-            "filters should be applied, and should discriminate"))
+    (with-redefs [log* (fn [& _] nil)]
+      ;; avoid unnecessary verbosity
+      (let [response (search-text entity "2607:f0d0:1002:0051:0000:0000:0000:0004")]
+        (is (= 400 (:status response)))))
 
-      (let [{:keys [status parsed-body]} (get search-uri
-                                              :headers {"Authorization" "45c1f5e3f05d0"}
-                                              :query-params {"query" query
-                                                             "tlp" "green"})]
-        (is (= 200 status))
-        (is (= 1 (count parsed-body))
-            "filters are applied, and match properly")))))
+    (let [response (search-raw entity {"query" query
+                                       "tlp" "red"})]
+      (is (= 200 (:status response)))
+      (is (empty? (:parsed-body response))
+          "filters should be applied, and should discriminate"))
+
+    (let [{:keys [status parsed-body]} (search-raw entity {:query query
+                                                           :tlp "green"})
+          matched-fields {:tlp "green"
+                          (keyword query-field) query}]
+      (is (= 200 status))
+      (is (<= 1 (count parsed-body)))
+      (is (every? #(= (select-keys % [(keyword query-field) :tlp])
```

Comment:
This `every?` looks like it'd report not very expressively, if the tests failed.

i.e. it'd report `(not (every? ...)`

`doseq` would be free of the gotcha.